volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,207 @@
+    cap_bset: interfaces.objects.ObjectInterface
+    cap_ambient: interfaces.objects.ObjectInterface
+
+    def astuple(self) -> Tuple:
+        """Returns a shallow copy of the capability sets in a tuple.
+
+        Otherwise, when dataclasses.astuple() performs a deep-copy recursion on
+        ObjectInterface will take a substantial amount of time.
+        """
+        return tuple(getattr(self, field.name) for field in fields(self))
+
+
+class Capabilities(plugins.PluginInterface):
+    """Lists process capabilities"""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pids",
+                description="Filter on specific process IDs.",
+                element_type=int,
+                optional=True,
+            ),
+        ]
+
+    def _check_capabilities_support(
+        self,
+        context: interfaces.context.ContextInterface,
+        vmlinux_module_name: str,
+    ):
+        """Checks that the framework supports at least as much capabilities as
+        the kernel being analysed. Otherwise, it shows a warning for the
+        developers.
+        """
+
+        vmlinux = context.modules[vmlinux_module_name]
+
+        try:
+            kernel_cap_last_cap = vmlinux.object_from_symbol(symbol_name="cap_last_cap")
+        except exceptions.SymbolError:
+            # It should be a kernel < 3.2
+            return
+
+        vol2_last_cap = extensions.kernel_cap_struct.get_last_cap_value()
+        if kernel_cap_last_cap > vol2_last_cap:
+            vollog.warning(
+                "Developers: The supported Linux capabilities of this plugin are outdated for this kernel"
+            )
+
+    @staticmethod
+    def _decode_cap(cap: interfaces.objects.ObjectInterface) -> str:
+        """Returns a textual representation of the capability set.
+        The format is a comma-separated list of capabilitites. In order to
+        summarize the output and if all the capabilities are enabled, instead of
+        the individual capabilities, the special name "all" will be shown.
+
+        Args:
+            cap: Kernel capability object. Usually a 'kernel_cap_struct' struct
+
+        Returns:
+            str: A string with a comma separated list of decoded capabilities
+        """
+        if isinstance(cap, renderers.NotAvailableValue):
+            return cap
+
+        cap_value = cap.get_capabilities()
+        if cap_value == 0:
+            return "-"
```

Comment:
This should be a `BaseAbsentValue` derived instance please (and can also be `if not cap_value`) just so that UIs that might want to know that there's no real data there can display it differently if needed

Reply:
In this case, the "-" (or the cap_value == 0) means that there is no capability enabled. It doesn't seem that any of the UnreadableValue, UnparsableValue, NotApplicableValue, NotAvailableValue are correct here. It should show an empty value in this case, it's not an error. What do you thing it could be the best option? I can also replace it with an empty string if you think it's better.

Reply:
Ok, I pushed a new version with an empty string. If you like we leave it like that.

Reply:
Hehehe, touche, you got me there, it doesn't really match any of the existing ones.  Given that's the case I'm happy with a `""` or a `"-"` or even a more descriptive `"No capabilities"` or something?  I'm happy for you to pick (you might be tired of having to change it, so don't worry "" is fine if you want!)  It's getting late here, so I'll probably push it tomorrow, feel free to update it or not before then...  5:)

Reply:
Haha, thanks. That should be everything. Also, all the test passed youhuu, so be free to check it again and merge it when ever you want.